bin/helpers/buildArtifacts.js

```diff
@@ -0,0 +1,212 @@
+          fs.unlinkSync(tmpFilePath);
+          resolve(true);
+        }
+        //no need to call the reject here, as it will have been called in the
+        //'error' stream;
+      });
+    });
+  });
+}
+
+const unzipFile = async (filePath, fileName) => {
+  return new Promise( async (resolve, reject) => {
+    await unzipper.Open.file(path.join(filePath, fileName))
+      .then(d => d.extract({path: filePath, concurrency: 5}))
+      .catch((err) => reject(err));
+    resolve();
+  });
+}
+
+const sendUpdatesToBstack = async (bsConfig, buildId, args, options) => {
+  let url = `${config.buildUrl}${buildId}/build_artifacts/status`;
+
+  let cypressJSON = utils.getCypressJSON(bsConfig);
+
+  let reporter = null;
+  if(!utils.isUndefined(args.reporter)) {
+    reporter = args.reporter;
+  } else if(cypressJSON !== undefined){
+    reporter = cypressJSON.reporter;
+  }
+
+  let data = {
+    feature_usage: {
+      downloads: {
+        eligible_download_folders: Number(process.env.BUILD_ARTIFACTS_TOTAL_COUNT),
+        successfully_downloaded_folders: Number(process.env.BUILD_ARTIFACTS_TOTAL_COUNT) - Number(process.env.BUILD_ARTIFACTS_FAIL_COUNT)
+      },
+      reporter: reporter
+    }
+  }
+
+  try {
+    await axios.post(url, data, options);
+  } catch (err) {
+    utils.sendUsageReport(bsConfig, args, err, Constants.messageTypes.ERROR, 'api_failed_build_artifacts_status_update');
+  }
+}
+
+exports.downloadBuildArtifacts = async (bsConfig, buildId, args) => {
+  process.env.BUILD_ARTIFACTS_FAIL_COUNT = 0;
+  process.env.BUILD_ARTIFACTS_TOTAL_COUNT = 0;
+
+  let url = `${config.buildUrl}${buildId}/build_artifacts`;
+  let options = {
+    auth: {
+      username: bsConfig.auth.username,
+      password: bsConfig.auth.access_key,
+    },
+    headers: {
+      'User-Agent': utils.getUserAgent(),
+    },
+  };
+
+  let message = null;
+  let messageType = null;
+  let errorCode = null;
+
+  try {
+      const res = await axios.get(url, options);
+      let buildDetails = res.data;
+
+      await createDirectories(buildId, buildDetails);
+      await parseAndDownloadArtifacts(buildId, buildDetails);
+
+      if (process.env.BUILD_ARTIFACTS_FAIL_COUNT > 0) {
+        messageType = Constants.messageTypes.ERROR;
+        message = Constants.userMessages.DOWNLOAD_BUILD_ARTIFACTS_FAILED.replace('<build-id>', buildId).replace('<machine-count>', process.env.BUILD_ARTIFACTS_FAIL_COUNT);
+        logger.error(message);
+      } else {
+        messageType = Constants.messageTypes.SUCCESS;
+        message = Constants.userMessages.DOWNLOAD_BUILD_ARTIFACTS_SUCCESS.replace('<build-id>', buildId).replace('<user-path>', process.cwd());
+        logger.info(message);
+      }
+
+      await sendUpdatesToBstack(bsConfig, buildId, args, options);
+      utils.sendUsageReport(bsConfig, args, message, messageType, null);
+  } catch (err) {
+    messageType = Constants.messageTypes.ERROR;
+    errorCode = 'api_failed_build_artifacts';
```

Comment:
Fetch this from utils.getErrorCodeFromMsg and fetch it from there.

Reply:
discussed f2f. not needed